### PR TITLE
elastic and subgraph replacement

### PR DESCRIPTION
**Pull request by AdriGeorge**
work with:

- ocean node enterprise: https://github.com/OceanProtocolEnterprise/ocean-node
- ocean.js: https://github.com/oceanprotocol/ocean.js/pull/1858 for other chains (for local with barge it works also without)

changes:

- update query
- update transform query result
- removed unused code

- replace subgraph functions with functions that call ocean node:
  - get opcData
  - get fees (publish)
  - getApprovedTokens (publish)
  - getUserOrders (profile)
  - get compute (when click on a service of dataset)
  - update querys elastic with correct index
  - removed history metadata updated (subgraph interaction)
  - removed unused subgraph functions

To run:
- start barge
- start node
- update env after barge started with: `npm run set-barge-env .`
- start market

### CODE REVIEW

File: src/@context/UserPreferences.tsx [hunk 1]
```diff
@@ -60,9 +60,7 @@ function UserPreferencesProvider({
   )
   const [locale, setLocale] = useState<string>()
   const [bookmarks, setBookmarks] = useState(localStorage?.bookmarks || [])
-  const [chainIds, setChainIds] = useState(
-    localStorage?.chainIds || appConfig.chainIds
-  )
```
Comment (FilipMasar) [inline src/@context/UserPreferences.tsx:65]: why? Maybe in localStorage you have your preferences stored

File: src/@types/AssetExtended.d.ts [hunk 2]
```diff
@@ -32,5 +32,7 @@ declare global {
     metadata: MetadataExtended
     services: ServiceExtended[]
     offchain?: OffChain // TODO - in future it will be directly included in Asset type in @oceanprotocol/lib
+    _id: string
+    _source: any
```
Comment (FilipMasar) [inline src/@types/AssetExtended.d.ts:36]: What are these? Why do we need it here? Maybe add some comments so it's easy to understand

**Comment by FilipMasar** [comment]
@AdriGeorge I tested it locally with
- barge `main` branch
- ocean-node community `main` branch

I published a new free dataset. I am able to see the detail of the dataset http://localhost:8008/asset/did:op:12eebc8defcc8f31e964490921a7f737ac9391c9a99df1bfbe97a710603b8e20
but I am not able to see the dataset in Catalogue, nor in my profile.

**Comment by AdriGeorge** [comment]
> @AdriGeorge I tested it locally with
> 
> * barge `main` branch
> * ocean-node community `main` branch
> 
> I published a new free dataset. I am able to see the detail of the dataset http://localhost:8008/asset/did:op:12eebc8defcc8f31e964490921a7f737ac9391c9a99df1bfbe97a710603b8e20 but I am not able to see the dataset in Catalogue (http://localhost:8008/search?sort=nft.created&sortOrder=desc), nor in my profile (http://localhost:8008/profile)

ok, fixed, try again please

**Review by FilipMasar — APPROVED** [review]
Good job. It works now

**Pull request merged**